ZenPacks/zenoss/DeviceSearch/device/adapters.py

```diff
@@ -48,20 +47,38 @@ def getSearchResults(self, parsedQuery, sorter=None, category=None, countOnly=Fa
             return
 
         def listMatchGlob(op, index, list):
-            return op(*[ MatchGlob(index, '*%s*' % i ) for i in list ])
+            return op(*[MatchGlob(index, '*%s*' % i) for i in list])
+
         dmd = self._dmd
-        kw_query = Or(listMatchGlob(And, 'titleOrId', keywords),
-                      listMatchGlob(And, 'getDeviceIp', keywords))
-        # Rank devices whose name match the query higher than other stuff
-        ranker = RankByQueries_Sum( (listMatchGlob(Or, 'titleOrId', keywords), 10), )
-        full_query = kw_query
-        cat = dmd.Devices.deviceSearch
-
-        querySet = full_query
-
-        catalogItems = cat.evalAdvancedQuery(querySet,  (ranker,) )
-        brainResults = [DeviceSearchResult(catalogItem)
-                        for catalogItem in catalogItems if checkPermission("View", catalogItem.getObject())]
+
+        try:
+            from Products.Zuul.catalog.interfaces import IModelCatalogTool
```

Comment:
it would look cleaner if we create two methods, one get_bla_from_model_catalog and get_bla_from_device_catalog